source/solvers/heat_transfer.cc

```diff
@@ -348,6 +348,11 @@ HeatTransfer<dim>::setup_assemblers()
     std::make_shared<HeatTransferAssemblerRobinBC<dim>>(
       this->simulation_control, simulation_parameters.boundary_conditions_ht));
 
+  // Neumann boundary condition
```

Comment:
The best things to do for these BCs would be to add the assemblers only if the BC type exists, but maybe that can be another PR.

Reply:
I will do it in this one.

Reply:
I choose to define a "has_type_bc" boolean to be "true" when the parameter is called but I don't know if this is the cleanest approach.

Reply:
Well that's not a bad approach. It's easy to understand and boundary conditions don't change type during simulations so it makes sense.